「ユドナリウムのキャラコマを作るやつ」へのサタスペ用コマ作成機能の追加

applehorn opened the pull request:
# 概要
下記キャラクターシート倉庫に登録されているキャラクターのコマを作成する機能に関して、
サタスペへの対応を行いました。
https://character-sheets.appspot.com/satasupe/

下記、生成されたコマのサンプルとなります。
[xml_sample.zip](https://github.com/blhsrwznrghfzpr/udonarium/files/5080540/xml_sample.zip)

# バグ（？）
以下のように項目に何も入っていない場合、ちょっと表示がいまいちな感じになります。
（他のシノビガミ等のキャラコマ作成でも同じになるようです）
<img src="https://user-images.githubusercontent.com/43114908/90334909-e1749f00-e00b-11ea-9000-de5bfb4cefca.jpg" width=600px>

# 備考
サタスペのコマ生成機能が欲しく、PR送らせていただきました。
他の同様のキャラシート倉庫からのコマ生成部分（シノビガミ等）を参考にして作成しましたが、
問題がありそうでしたらリジェクト等の対応していただいて問題ありません。
よろしくおねがいします。

applehorn committed:
update plugins/charazip for satasupe

blhsrwznrghfzpr reviewed (commented):
ありがとうございます。
何点か気になるところがありました。修正をお願いできますか？
もし面倒でしたらこちらで修正を行います。

- [x] 趣味がうまく取得できていないように思えます。
sampleのキャラクターシートでは「アラサガシ」「パフォーマンス」「宗教」なのに対して、
生成されたコマでは「サビシガリヤ」「カワイイ」「ハイソ」になっています。
- [x] 「状態」以降の項目で`null`と出てきてしまってます。プログラムに馴染みのない人には不親切なので、`null`はなるべく表示されないようにしたいです

表示がイマイチになってしまうのは認識していましたが、放置していただけなので気にしなくてもいいです

applehorn committed:
Fixed the following
 - Fixed a mistake in getting hobby items
 - Null -> blank after "Status"

applehorn commented:
確認ありがとうございます！
ご指摘いただいた二点を修正いたしました。
よろしくおねがいします。

blhsrwznrghfzpr commented on src/app/plugins/charazip/system/satasupe.ts:
```diff
@@ -0,0 +1,347 @@
+(教養)R>={難易度}[{目標成功度},{ファンブル値}]
+(戦闘)R>={難易度}[{目標成功度},{ファンブル値}]
+(肉体)R>={難易度}[{目標成功度},{ファンブル値}]
+(精神)R>={難易度}[{目標成功度},{ファンブル値}]
```
ユドナリウムでのチャットパレットは`{タグ名}`でキャラクターシートの`タグ名`の中身を置き換えできるようになっています。
このままだと`(精神)R>=[,]`と表示されるだけになってしまいますが、その想定で問題ないでしょうか？
キャラクターシートに設定されてる値を引っ張りたい場合は`{精神}`のようにすると良いです。

また、難易度や目標成功度、ファンブル値などは逐次手入力してもらう想定でしょうか？
その場合は何も書かずに `{精神}R>=`とだけしておく方が、文字を消す手間が入らなくていいと思います。

blhsrwznrghfzpr commented on src/app/plugins/charazip/system/satasupe.ts:
```diff
@@ -0,0 +1,347 @@
+    const emotionElement = gameCharacter.createDataElement('性業値', '');
+    gameCharacter.detailDataElement.appendChild(emotionElement);
+    emotionElement.appendChild(
+      gameCharacter.createDataElement('性業値', json.base.emotion)
```
「性業値」という同じタグ名でElementを2つ作っているので、チャットパレット内の`SR{性業値}`がうまく引用できなくなってしまっています。
親の`emotionElement`のタグ名を別の名前に変えるといいと思います。

blhsrwznrghfzpr reviewed (commented):
修正ありがとうございます。チャットパレットで気になる点があったので確認お願いします。

applehorn committed:
Fix Chat Palette
･Erase Difficulty, Target Value, Fumble Value, Deadly Value
･Change emotionElement　”性業値” -> "性業"

applehorn commented:
> このままだと(精神)R>=[,]と表示されるだけになってしまいますが、その想定で問題ないでしょうか？
> キャラクターシートに設定されてる値を引っ張りたい場合は{精神}のようにすると良いです。

値は引っ張って欲しいので、{精神}の形で修正しました。
また、システム的には({精神}の値＋n)で判定することができるので、({精神})R>=の表記に統一しました。

>また、難易度や目標成功度、ファンブル値などは逐次手入力してもらう想定でしょうか？

こちらはその想定ですので、空にしておきました。

>親のemotionElementのタグ名を別の名前に変えるといいと思います。

親のemotionElementのタグ名を”性業”に修正いたしました。

blhsrwznrghfzpr committed:
fix: 行頭の空白除去

blhsrwznrghfzpr committed:
chore: システムを辞書順に並び替え

blhsrwznrghfzpr commented:
マージしました。寄稿ありがとうございます！
